intel-analytics/BigDL
# Scale and Mask

## What changes were proposed in this pull request?
1. added 5 fields into protobuf definition, inputDimMasks, inputScales, outputDimMasks, outputScales, isMklInt8Enabled; updated related auto-generated code.
2. added 1 Trait MklInt8Convertible which hold data and methods to calculate required information for converting from fp32 to int8.
3. added functions to serialize and deserialize MKL-DNN Int8 related values 
4. made MKL-DNN Linear module inherit MKL-DNN Int8 Trait
(Please fill in changes proposed in this patch)

## How was this patch tested?
1. Construct a module, Linear in my case.
2. Generate and hold random values
2. Assign random values to above fields
3. Save the module
4. Load the module
5. Verify the values stored in the fields are the same as assigned at step #2.

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If it is possible, please attach a screenshot; otherwise, remove this)

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX
